File: Programming - 2/Lab_5_6/Shapes/Shape.cs
using System;

namespace Shapes
{
    public abstract class Shape
    {
        public abstract double Perimeter { get; }
        public abstract double Area { get; }
        protected static double GetDistance((double x, double y) A, (double x, double y) B)
            => Math.Sqrt((A.x - B.x) * (A.x - B.x) + (A.y - B.y) * (A.y - B.y));
    }
}

File: Programming - 2/Lab_5_6/Lab_6_CPP/Back/Shape.cpp
#include <cmath>
#include "Shape.h"

double Shape::GetDistance(Point2D A, Point2D B)
{
	return sqrt((A.x - B.x) * (A.x - B.x) + (A.y - B.y) * (A.y - B.y));
}
